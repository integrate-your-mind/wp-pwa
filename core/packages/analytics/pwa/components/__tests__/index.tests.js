import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import { types } from 'mobx-state-tree';
import Analytics from '..';

const Stores = types.model('Stores').props({
  build: types.frozen({
    isPwa: true,
    channel: 'pwa',
    packages: 'saturn-theme,wp-org-connection',
  }),
  connection: types.frozen({
    selectedItem: { type: 'post', id: 60 },
  }),
  analytics: types.frozen({
    customDimensions: () => ({
      dimension1: 'dim1_post60',
      dimension2: 'dim2_post60',
    }),
    googleAnalytics: {
      ids: ['UA-12345678-1', 'UA-12345678-2'],
    },
    googleTagManager: {
      ids: ['GTM-112233'],
      clientProperties: {
        anonymize: false,
        siteId: 'site1122334455',
        userIds: ['user00', 'user01'],
        theme: 'saturn-theme',
        extensions: 'saturn-theme,wp-org-connection',
        plan: 'enterprise',
        pageType: 'pwa',
        dev: true,
      },
      pageViewProperties: {
        site: 'https://demo.frontity.test',
        title: 'The Beauties of Gullfoss â€“ Demo Frontity',
        url: 'https://demo.frontity.test/the-beauties-of-gullfoss/',
        type: 'post',
        id: 60,
        format: 'pwa',
        route: 'single',
        hash: 'ZO0H2I9kJ0arOdstZdG',
        customDimensions: {
          dimension1: 'dim1_post60',
          dimension2: 'dim2_post60',
        },
      },
    },
    comScore: {
      ids: ['test1', 'test2'],
    },
  }),
});

describe('Analytics > PWA Components', () => {
  it('renders correctly', () => {
    const stores = Stores.create();
    const tree = renderer.create(
      <Provider stores={stores}>
        <Analytics />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
