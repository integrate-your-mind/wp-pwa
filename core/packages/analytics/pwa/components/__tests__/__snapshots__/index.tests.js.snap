// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics > PWA Components renders correctly 1`] = `
Array [
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
  'gtm.start': Date.now(),
  event: 'gtm.js',
});
window.dataLayer.push({
  event: 'wpPwaProperties',
  wpPwaProperties: {
  \\"anonymize\\": false,
  \\"siteId\\": \\"site1122334455\\",
  \\"userIds\\": [
    \\"user00\\",
    \\"user01\\"
  ],
  \\"theme\\": \\"saturn-theme\\",
  \\"extensions\\": \\"saturn-theme,wp-org-connection\\",
  \\"plan\\": \\"enterprise\\",
  \\"pageType\\": \\"pwa\\",
  \\"dev\\": true
},
});
window.dataLayer.push({
  event: 'virtualPageview',
  virtualPageview: {
  \\"site\\": \\"https://demo.frontity.test\\",
  \\"title\\": \\"The Beauties of Gullfoss â€“ Demo Frontity\\",
  \\"url\\": \\"https://demo.frontity.test/the-beauties-of-gullfoss/\\",
  \\"type\\": \\"post\\",
  \\"id\\": 60,
  \\"format\\": \\"pwa\\",
  \\"route\\": \\"single\\",
  \\"hash\\": \\"ZO0H2I9kJ0arOdstZdG\\",
  \\"customDimensions\\": {
    \\"dimension1\\": \\"dim1_post60\\",
    \\"dimension2\\": \\"dim2_post60\\"
  }
},
});",
      }
    }
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-12345678-1', 'auto', 'tracker_UA_12345678_1');
ga('tracker_UA_12345678_1.send', 'pageview', {\\"dimension1\\":\\"dim1_post60\\",\\"dimension2\\":\\"dim2_post60\\"});",
      }
    }
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-12345678-2', 'auto', 'tracker_UA_12345678_2');
ga('tracker_UA_12345678_2.send', 'pageview', {\\"dimension1\\":\\"dim1_post60\\",\\"dimension2\\":\\"dim2_post60\\"});",
      }
    }
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
var _comscore = _comscore || [];
_comscore.push({ c1: \\"2\\", c2: \\"test1\\" });
(function() {
  var s = document.createElement(\\"script\\"), el = document.getElementsByTagName(\\"script\\")[0]; s.async = true;
  s.src = (document.location.protocol == \\"https:\\" ? \\"https://sb\\" : \\"http://b\\") + \\".scorecardresearch.com/beacon.js\\";
  el.parentNode.insertBefore(s, el);
})();",
      }
    }
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
var _comscore = _comscore || [];
_comscore.push({ c1: \\"2\\", c2: \\"test2\\" });
(function() {
  var s = document.createElement(\\"script\\"), el = document.getElementsByTagName(\\"script\\")[0]; s.async = true;
  s.src = (document.location.protocol == \\"https:\\" ? \\"https://sb\\" : \\"http://b\\") + \\".scorecardresearch.com/beacon.js\\";
  el.parentNode.insertBefore(s, el);
})();",
      }
    }
  />,
]
`;
